tslearn/docs/examples/plot_dtw_custom_metric.py

```diff
@@ -0,0 +1,152 @@
+distances between the multidimensional boolean arrays. In both cases, the
+corresponding time series are represented at the left and at the top of each
+cost matrix.
+
+The optimal path, that is the path that minimizes the total user-defined cost
+from the first time point to the last one, is represented in white on the
+image.
+
+.. [1] H. Sakoe and S. Chiba, "Dynamic programming algorithm optimization
+       for spoken word recognition". IEEE Transactions on Acoustics, Speech,
+       and Signal Processing, 26(1), 43-49 (1978).
+
+.. [2] Definition of the length of an arc on `Wikipedia
+       <https://en.wikipedia.org/wiki/Arc_(geometry)#Length_of_an_arc_of_a_circle>`_.
+
+.. [3] See Hammig distance in `Scipy's documentation
+       <https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.hamming.html>`_.
+
+"""
+
+# Author: Romain Fayat
+# License: BSD 3 clause
+
+import numpy as np
+from numpy import pi
+from sklearn.metrics import pairwise_distances
+import matplotlib.pyplot as plt
+from matplotlib.colors import LinearSegmentedColormap
+
+from tslearn.generators import random_walks
+from tslearn import metrics
+from tslearn.preprocessing import TimeSeriesScalerMeanVariance
+
+np.random.seed(0)
+n_ts, sz = 2, 100
+
+
+# Example 1 : Length of the arc between two angles on a circle
+def arc_length(angle_1, angle_2, r=1.):
+    "Length of the arc between two angles (in rad) on a circle of radius r."
+    # Compute the angle between the two inputs between 0 and 2*pi.
+    theta = np.mod(angle_2 - angle_1, 2*pi)
+    if theta > pi:
+        theta = theta - 2 * pi
+    # Return the length of the arc
+    L = r * np.abs(theta)
+    return(L)
+
+dataset_1 = random_walks(n_ts=n_ts, sz=sz, d=1)
+scaler = TimeSeriesScalerMeanVariance(mu=0., std=pi)  # Rescale the time series
+dataset_scaled_1 = scaler.fit_transform(dataset_1)
+
+# DTW using a function as the metric argument
+path_1, sim_1 = metrics.dtw_path_from_metric(
+    dataset_scaled_1[0], dataset_scaled_1[1], metric=arc_length
+)
+
+# Example 2 : Hamming distance between 2 multi-dimensional boolean time series
+rw = random_walks(n_ts=n_ts, sz=sz, d=15, std=.3)
+dataset_2 = np.mod(np.floor(rw), 4) == 0
+
+# DTW using one of the options of sklearn.metrics.pairwise_distances
+path_2, sim_2 = metrics.dtw_path_from_metric(
+    dataset_2[0], dataset_2[1], metric="hamming"
+)
+
+# Plots
+# Grab the data for the plots
+paths = [path_1, path_2]
+data = [dataset_scaled_1, dataset_2]
+distances_1 = pairwise_distances(
+    dataset_scaled_1[0], dataset_scaled_1[1], metric=arc_length
+)
+distances_2 = pairwise_distances(dataset_2[0], dataset_2[1], metric="hamming")
+distances = [distances_1, distances_2]
+
+# Definitions for the axes
+left, bottom = [.01, .51], .1
+w_ts, h_ts = .1, .2
+left_h = [l + w_ts + .01 for l in left]
+width, height = .325, .65
+bottom_h = bottom + height + .02
+
+# Create the axes and plot the distance matrices
+plt.figure(1, figsize=(12, 6))
```

Comment:
This example is great. I have maybe one minor remark: it would be better to generate two distinct figures and pick the second one as the thumbnail in the gallery, as done there: https://github.com/rtavenar/tslearn/blob/master/tslearn/docs/examples/metrics/plot_lb_keogh.py